engine/src/main/coffee/extensions/palette.coffee

```diff
@@ -0,0 +1,2340 @@
+      [	153	,	216	,	201	]	,
+      [	102	,	194	,	164	]	,
+      [	65	,	174	,	118	]	,
+      [	35	,	139	,	69	]	,
+      [	0	,	88	,	36	]	],
+    [	[	247	,	252	,	253	]	,
+      [	229	,	245	,	249	]	,
+      [	204	,	236	,	230	]	,
+      [	153	,	216	,	201	]	,
+      [	102	,	194	,	164	]	,
+      [	65	,	174	,	118	]	,
+      [	35	,	139	,	69	]	,
+      [	0	,	109	,	44	]	,
+      [	0	,	68	,	27	]	]]
+
+  Greens = [
+    [	[	229	,	245	,	224	]	,
+      [	161	,	217	,	155	]	,
+      [	49	,	163	,	84	]	],
+    [	[	237	,	248	,	233	]	,
+      [	186	,	228	,	179	]	,
+      [	116	,	196	,	118	]	,
+      [	35	,	139	,	69	]	],
+    [	[	237	,	248	,	233	]	,
+      [	186	,	228	,	179	]	,
+      [	116	,	196	,	118	]	,
+      [	49	,	163	,	84	]	,
+      [	0	,	109	,	44	]	],
+    [	[	237	,	248	,	233	]	,
+      [	199	,	233	,	192	]	,
+      [	161	,	217	,	155	]	,
+      [	116	,	196	,	118	]	,
+      [	49	,	163	,	84	]	,
+      [	0	,	109	,	44	]	],
+    [	[	237	,	248	,	233	]	,
+      [	199	,	233	,	192	]	,
+      [	161	,	217	,	155	]	,
+      [	116	,	196	,	118	]	,
+      [	65	,	171	,	93	]	,
+      [	35	,	139	,	69	]	,
+      [	0	,	90	,	50	]	],
+    [	[	247	,	252	,	245	]	,
+      [	229	,	245	,	224	]	,
+      [	199	,	233	,	192	]	,
+      [	161	,	217	,	155	]	,
+      [	116	,	196	,	118	]	,
+      [	65	,	171	,	93	]	,
+      [	35	,	139	,	69	]	,
+      [	0	,	90	,	50	]	],
+    [	[	247	,	252	,	245	]	,
+      [	229	,	245	,	224	]	,
+      [	199	,	233	,	192	]	,
+      [	161	,	217	,	155	]	,
+      [	116	,	196	,	118	]	,
+      [	65	,	171	,	93	]	,
+      [	35	,	139	,	69	]	,
+      [	0	,	109	,	44	]	,
+      [	0	,	68	,	27	]	]]
+
+  Greys = [
+    [	[	240	,	240	,	240	]	,
+      [	189	,	189	,	189	]	,
+      [	99	,	99	,	99	]	],
+    [	[	247	,	247	,	247	]	,
+      [	204	,	204	,	204	]	,
+      [	150	,	150	,	150	]	,
+      [	82	,	82	,	82	]	],
+    [	[	247	,	247	,	247	]	,
+      [	204	,	204	,	204	]	,
+      [	150	,	150	,	150	]	,
+      [	99	,	99	,	99	]	,
+      [	37	,	37	,	37	]	],
+    [	[	247	,	247	,	247	]	,
+      [	217	,	217	,	217	]	,
+      [	189	,	189	,	189	]	,
+      [	150	,	150	,	150	]	,
+      [	99	,	99	,	99	]	,
+      [	37	,	37	,	37	]	],
+    [	[	247	,	247	,	247	]	,
+      [	217	,	217	,	217	]	,
+      [	189	,	189	,	189	]	,
+      [	150	,	150	,	150	]	,
+      [	115	,	115	,	115	]	,
+      [	82	,	82	,	82	]	,
+      [	37	,	37	,	37	]	],
+    [	[	255	,	255	,	255	]	,
+      [	240	,	240	,	240	]	,
+      [	217	,	217	,	217	]	,
+      [	189	,	189	,	189	]	,
+      [	150	,	150	,	150	]	,
+      [	115	,	115	,	115	]	,
+      [	82	,	82	,	82	]	,
+      [	37	,	37	,	37	]	],
+    [	[	255	,	255	,	255	]	,
+      [	240	,	240	,	240	]	,
+      [	217	,	217	,	217	]	,
+      [	189	,	189	,	189	]	,
+      [	150	,	150	,	150	]	,
+      [	115	,	115	,	115	]	,
+      [	82	,	82	,	82	]	,
+      [	37	,	37	,	37	]	,
+      [	0	,	0	,	0	]	]]
+
+module.exports = {
+
+  init: (workspace) ->
+
+    getMyColor = () ->
+      self = workspace.selfManager.self()
+      mycolor = self.getVariable("color")
+      if(mycolor is undefined) # called by a patch
```

Comment:
You should be able to do `self instanceof Patch` to check if something is a patch to avoid the extra `getVariable()` call.  Same with `setMyColor()`

Reply:
Patch is not defined here, is there something I should import?

Reply:
Yeah, you can add a require statement for it, [like this](https://github.com/NetLogo/Tortoise/blob/master/engine/src/main/coffee/engine/prim/listprims.coffee#L5) but you'll have to figure out the relative path to use.